AddQtAndroidApk.cmake

```diff
@@ -1,8 +1,7 @@
-cmake_minimum_required(VERSION 3.0)
-cmake_policy(SET CMP0026 OLD) # allow use of the LOCATION target property
+cmake_minimum_required(VERSION 3.0.0 FATAL_ERROR)
 
 # store the current source directory for future use
-set(QT_ANDROID_SOURCE_DIR ${CMAKE_CURRENT_LIST_DIR})
+set(QT_ANDROID_SOURCE_DIR ${CMAKE_CURRENT_LIST_DIR} CACHE STRING "Source directory of AddQtAndroidApk.cmake")
```

Comment:
I'm not a huge fan of adding this (and the others) paths to the cache, it means that they can be edited, which is unnecessary if the user configured its build properly (as described in readme).

Moreover, it adds another set of variables which are more or less aliases to already defined variables, which
in my opinion adds confusion.

And to finish, some of these variables are really meant for internal use only (this one, for example), I don't really see any benefit of exposing them like this.

Reply:
I will try to answer to each of your review.
Sorry for some obscure commits.
Main reason for cache variables, the CMakeLists.txt file adding the script etc... is to be able to use CMake's [FetchContent](https://cmake.org/cmake/help/v3.11/module/FetchContent.html).
Like that in my code i can simply
```
FetchContent_Declare(
    QtAndroidCMake
    GIT_REPOSITORY https://github.com/OlivierLDff/QtAndroidCMake
    GIT_TAG        master
)
FetchContent_MakeAvailable(QtAndroidCMake)
```
The FetchContent_MakeAvailable call "out of the box" the CMakeLists.txt file if any is present. So in my script i have one line to declare the dependency and one to download it. I don't need know exactly the name of the file `*.cmake`. This may seem like a simple answer but it helps when reading my script. Each script i might need is "inserted" into the project the same way.
An example that might work out of the box : https://github.com/OlivierLDff/QaterialGallery/blob/master/platform/PostBuildPlatform.cmake
You can see in there that i created other script QtWindowsCMake and QtIosCMake with the same philosophy.
So long story short, using that `MakeAvailable` function, the variable defined before the macro needs to be stored somewhere when the macro call happen.
Maybe there would be a way to scope them to the file? I added them in the cache as a quick workaround to be able to use MakeAvailable.

Reply:
I see.

That's still confusing to see them and be able to edit them, but if that's the only way then I'll be fine with that.

Reply:
It was a quick fix so maybe there is a better way to handle that

Reply:
A good idea to fix this issue would be to go back the way you were declaring the variable. And store them in cache with same named variables in the CMakeLists.txt. This solution will result in:
* If user simply include the script then no variable get stored in cache
* If user use FetchContent to include the script then variables are going to be stored in cache.
Would you consider this as a "fix"?

Reply:
That would not be perfect, but definitely better, yes :+1:

We should also explicitly mention in the readme file that CMakeLists.txt is meant for this specific use case of FetchContent; for regular use, one should still include AddQtAndroidApk.cmake.

Reply:
I did it in 372ebda76952b30d29c4f1c63f31d2d559637c8e
Where do you think i should mention the use case in the readme?
Just before the How to run CMake header?
Alternative: How to integrate using FetchContent
Something like that?

Reply:
Hmm maybe we should not add anything to the readme, and add a comment about FetchContent directly inside the CMakeLists.txt file instead. Using FetchContent works out of the box with this file, right? If so, we don't need to add anything to the readme file about FetchContent.

Regarding the readme: you should also remove the first paragraph (*"this fork..."*), and put any relevant addition into the dedicated sections.

I would also remove the VERBOSE option and force it by default, like before. Again, let's aim at simplicity.

Reply:
I added the verbose option because it's an option that i exposed in lots of CMake scripts. CMake script are in general non verbose.